packages/rainbowkit/src/wallets/connectorsForWallets.ts

```diff
@@ -117,44 +118,36 @@ export const connectorsForWallets = (
       }
     }
 
-    const walletMetaData = (
-      // For now we should only use these as the additional parameters
-      additionalRkParams?: Pick<
-        WalletDetailsParams['rkDetails'],
-        'isWalletConnectModalConnector' | 'showQrModal'
-      >,
-    ) => {
-      return {
-        rkDetails: omitUndefinedValues({
-          ...walletMeta,
-          groupIndex,
-          groupName,
-          isRainbowKitConnector: true,
-          // These additional params will be used in rainbowkit react tree to
-          // merge `walletConnectWallet` and `walletConnect` connector from wagmi with
-          // showQrModal: true. This way we can let the user choose if they want to
-          // connect via QR code or open the official walletConnect modal instead
-          ...(additionalRkParams ? additionalRkParams : {}),
-        }),
-      };
+    const rainbowKitDetails = {
+      rkDetails: omitUndefinedValues({
+        ...walletMeta,
+        groupIndex,
+        groupName,
+        isRainbowKitConnector: true,
+      }),
     };
 
-    const isWalletConnectConnector = walletMeta.id === 'walletConnect';
-
-    if (isWalletConnectConnector) {
-      connectors.push(
-        createConnector(
-          walletMetaData({
-            isWalletConnectModalConnector: true,
-            showQrModal: true,
-          }),
-        ),
-      );
-    }
-
-    const connector = createConnector(walletMetaData());
+    const connector = createWagmiConnector((config) => ({
+      ...createRainbowKitConnector(config),
+      ...rainbowKitDetails,
+    }));
 
     connectors.push(connector);
+
+    if (walletMeta.id === 'walletConnect') {
+      const walletConnectConnector = createWagmiConnector((config) => ({
+        ...walletConnect({
+          projectId,
+          metadata: walletConnectMetaData,
+          ...walletConnectParameters,
+          showQrModal: true,
+        })(config),
+        ...rainbowKitDetails,
+        isWalletConnectModalConnector: true,
+      }));
+
+      connectors.push(walletConnectConnector);
+    }
```

Comment:
Creating new WalletConnect instance with `showQrModal: true` + append it's metadata.